com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AnimationSetBuilder.java

```diff
@@ -110,35 +117,162 @@ private void buildAnimations(Task<Void> task, AnimationSetDesc.Builder animSetDe
             ByteArrayInputStream animFileIS = new ByteArrayInputStream(animFile.getContent());
             AnimationSet.Builder animBuilder = AnimationSet.newBuilder();
             ArrayList<String> animationIds = new ArrayList<String>();
+
+            boolean isCollada = false;
+            String suffix = BuilderUtil.getSuffix(animFile.getPath());
+            if (suffix.equals("dae")) {
+                isCollada = true;
+            }
+
             try {
-                ColladaUtil.loadAnimations(animFileIS, animBuilder, animId, animationIds);
+                if (isCollada)
+                    loadColladaAnimations(animBuilder, animFileIS, animId, parentId);
+                else
+                    loadModelAnimations(bones, animBuilder, animFileIS, animId, parentId, animFile.getPath(), animationIds);
+
             } catch (XMLStreamException e) {
                 throw new CompileExceptionError(animFile, e.getLocation().getLineNumber(), "Failed to load animation: " + e.getLocalizedMessage(), e);
             } catch (LoaderException e) {
                 throw new CompileExceptionError(animFile, -1, "Failed to load animation: " + e.getLocalizedMessage(), e);
             }
 
             animationSetBuilder.addAllAnimations(animBuilder.getAnimationsList());
-            if(animBuilder.getBoneListList().size() > boneList.size()) {
-                boneList = animBuilder.getBoneListList();
+        }
+    }
+
+// For the Editor
+
+    static public AnimationSetDesc getAnimationSetDesc(InputStream animationSetStream) throws IOException {
+        InputStreamReader animSetDescISR = new InputStreamReader(animationSetStream);
+        AnimationSetDesc.Builder animSetDescBuilder = AnimationSetDesc.newBuilder();
+        TextFormat.merge(animSetDescISR, animSetDescBuilder);
+        return animSetDescBuilder.build();
+    }
+
+    private static ArrayList<String> makeUnique(List<String> items) {
```

Comment:
Is there any point in using a Set (for faster contains() lookups and then convert the Set to a List?

Also, in general I think we are usually too specific with the types in our functions in general. Why not return a List as opposed to an ArrayList? Is it important that it is an ArrayList specifically?

Reply:
THere won't be that many animations to be honest, but a Set is a good idea anyways.